File: C++/55 Jump Game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int len = nums.size();
        if(len <= 1) return true;
        int end = len-1;
        for(int i=len-2; i>=0; i--)
            if(i + nums[i] >= end) 
                end = i;
        return end == 0;
    }
};

File: 55 Jump Game.rs
/*
执行用时 :0 ms, 在所有 rust 提交中击败了100.00% 的用户
内存消耗 :2.3 MB, 在所有 rust 提交中击败了100.00%的用户
*/
impl Solution {
    pub fn can_jump(nums: Vec<i32>) -> bool {
        let len: i32 = nums.len() as i32;
        if(len <= 1) {return true;}
        let mut end: i32 = len - 1;
        for (i, x) in nums.iter().enumerate().rev() {
            if(i as i32 + x >= end) {
                end = i as i32;
            }
        }
        end == 0
    }
}